src/modules/sync/diff/transformToAddModel.ts

```diff
@@ -0,0 +1,157 @@
+import {
+  ContentTypeElements,
+  ContentTypeModels,
+  ContentTypeSnippetModels,
+  TaxonomyModels,
+} from "@kontent-ai/management-sdk";
+
+import {
+  assetAttributeName,
+  assetExternalIdAttributeName,
+  itemExternalIdLinkAttributeName,
+  itemLinkAttributeName,
+} from "../../../constants/richText.js";
+import { RequiredCodename } from "../../../utils/types.js";
+import { customAssetCodenameAttributeName, customItemLinkCodenameAttributeName } from "../../constants/syncRichText.js";
+import { ContentTypeSnippetsSyncModel, ContentTypeSyncModel, TaxonomySyncModel } from "../types/fileContentModel.js";
+import { SyncGuidelinesElement } from "../types/syncModel.js";
+
+export const transformTaxonomyToAddModel = (
+  taxonomy: TaxonomySyncModel,
+): RequiredCodename<TaxonomyModels.IAddTaxonomyRequestModel> => ({
+  ...taxonomy,
+  terms: taxonomy.terms.map(transformTaxonomyToAddModel),
+});
+
+export const transformTypeToAddModel =
+  (params: TransformElementParams) =>
+  (type: ContentTypeSyncModel): RequiredCodename<ContentTypeModels.IAddContentTypeData> => ({
+    ...type,
+    content_groups: type.content_groups?.slice(),
```

Comment:
nitpick: magic with slice => out of consistency I would make it the same as other arrays